### PR TITLE
Rewrite add-machine help text

### PR DESCRIPTION
**Pull request by timClicks**
### Checklist

 - [x] Checked if it requires a [pylibjuju](https://github.com/juju/python-libjuju) change?
 - [ ] <strike>Added [integration tests](https://github.com/juju/juju/tree/develop/tests) for the PR?</strike> N/A
 - [ ] <strike>Added or updated [doc.go](https://discourse.jujucharms.com/t/readme-in-packages/451) related to packages changed?</strike> N/A
 - [x] Do comments answer the question of why design decisions were made?

## Description of change

This change aims to improve the readability of 'juju help add-machine'. It breaks out the three modes of operation into separate sections, makes the language more consistent and documents all options. 

Some small changes

- all lines terminate before 80 chars
- remove unnecessary `init()` function
- migrate examples to syntax used by Kubernetes command docs

## QA steps

```sh
juju help add-machine
```

### CODE REVIEW

**Comment by hpidcock** [comment]
@timClicks ```2020-06-12T05:02:03.1078107Z ----------------------------------------------------------------------
2020-06-12T05:02:03.1078565Z ##[error]FAIL: main_test.go:839: MainSuite.TestAllCommandsPurpose
2020-06-12T05:02:03.1078838Z 
2020-06-12T05:02:03.1079038Z main_test.go:867:
2020-06-12T05:02:03.1079292Z     c.Check(strings.HasSuffix(purpose, "."), jc.IsTrue,
2020-06-12T05:02:03.1079540Z         comment("is missing full stop in Purpose"))
2020-06-12T05:02:03.1079762Z ... obtained bool = false
2020-06-12T05:02:03.1080205Z ... command "add-machine" is missing full stop in Purpose
2020-06-12T05:02:03.1080329Z 
2020-06-12T05:02:03.1145182Z 
2020-06-12T05:02:03.1145879Z ----------------------------------------------------------------------```

File: cmd/juju/machine/add.go [hunk 1]
```diff
@@ -120,18 +167,18 @@ type addCommand struct {
 func (c *addCommand) Info() *cmd.Info {
 	return jujucmd.Info(&cmd.Info{
 		Name:    "add-machine",
-		Args:    "[<container>:machine | <container> | ssh:[user@]host | winrm:[user@]host | placement]",
-		Purpose: "Start a new, empty machine and optionally a container, or add a container to a machine.",
+		Args:    "[<container-type>[:<machine-id>] | (ssh|winrm):[<user>@]<host> | <placement>]",
+		Purpose: "Provision a new machine or assign one to the model",
```
Comment (hpidcock) [inline cmd/juju/machine/add.go:171]: Just need a full stop at the end of this string.

Comment (timClicks) [reply cmd/juju/machine/add.go]: Fixed :)

**Comment by timClicks** [comment]
$$merge$$

File: cmd/juju/machine/add.go [hunk 2]
```diff
@@ -34,64 +34,111 @@ import (
...
+
+Add "placement directives" as an argument give Juju additional information 
+about how to allocate the machine in the cloud. For example, one can direct 
+the MAAS provider to acquire a particular node by specifying its hostname.
+
+
+Manual provisioning
+
+Call add-machine with the address of a network-accessible computer to 
+allocate that machine to the model. You should connect
```
Comment (SimonRichardson) [inline cmd/juju/machine/add.go:74]: > You should connect 

Leaving me hanging here.

**Pull request merged**